test/methods.jl

```diff
@@ -256,35 +256,134 @@ end
+            (geometry = (10.0, 0.1), test = 3)
+            (geometry = (10.0, 0.2), test = missing)
+        ])
+        # Extract a vector of polygons
+        ex = extract(rast_m, [p, p])
+        @test eltype(ex) == NamedTuple{(:geometry,:test)}
+        @test all(ex .=== [
+            (geometry = (9.0, 0.1), test = 1)
+            (geometry = (10.0, 0.1), test = 3)
+            (geometry = (10.0, 0.2), test = missing)
+            (geometry = (9.0, 0.1), test = 1)
+            (geometry = (10.0, 0.1), test = 3)
+            (geometry = (10.0, 0.2), test = missing)
+        ])
+        # Test all the keyword combinations
+        @test all(extract(rast_m, p) .=== [
+            (geometry = (9.0, 0.1), test = 1)
+            (geometry = (10.0, 0.1), test = 3)
+            (geometry = (10.0, 0.2), test = missing)
+        ])
+        @test all(extract(rast_m, p; geometry=false) .=== [
+            (test = 1,)
+            (test = 3,)
+            (test = missing,)
+        ])
+        @test all(extract(rast_m, p; geometry=false, index=true) .=== [
+            (index = CartesianIndex(1, 1), test = 1)
+            (index = CartesianIndex(2, 1), test = 3)
+            (index = CartesianIndex(2, 2), test = missing)
+        ])
+        @test all(extract(rast_m, p; index=true) .=== [
+             (geometry = (9.0, 0.1), index = CartesianIndex(1, 1), test = 1)
+             (geometry = (10.0, 0.1), index = CartesianIndex(2, 1), test = 3)
+             (geometry = (10.0, 0.2), index = CartesianIndex(2, 2), test = missing)
+        ])
+        @test extract(rast_m, p; skipmissing=true) == [
+            (geometry = (9.0, 0.1), test = 1)
+            (geometry = (10.0, 0.1), test = 3)
+        ]                                                         
+        @test extract(rast_m, p; skipmissing=true, geometry=false) == [
+            (test = 1,)
+            (test = 3,)
+        ]                                                         
+        @test extract(rast_m, p; skipmissing=true, geometry=false, index=true) == [
+            (index = CartesianIndex(1, 1), test = 1)
+            (index = CartesianIndex(2, 1), test = 3)
+        ]                                                         
+        @test extract(rast_m, p; skipmissing=true, index=true) == [
+            (geometry = (9.0, 0.1), index = CartesianIndex(1, 1), test = 1)
+            (geometry = (10.0, 0.1), index = CartesianIndex(2, 1), test = 3)
+        ]                                                         
+        # Empty geoms
+        @test extract(rast, []) == NamedTuple{(:geometry, :test),Tuple{Missing,Missing}}[]
+        @test extract(rast, []; geometry=false) == NamedTuple{(:test,),Tuple{Missing}}[]
+        # Missing coord errors
+        @test_throws ArgumentError extract(rast, [(0.0, missing), (9.0, 0.1), (9.0, 0.2), (10.0, 0.3)])
+        @test_throws ArgumentError extract(rast, [(X=0.0, Y=missing), (9.0, 0.1), (9.0, 0.2), (10.0, 0.3)])
```

Comment:
Add another test if an invalid geometry is not the first geometry

```suggestion
        @test_throws ArgumentError extract(rast, [(0.0, missing), (9.0, 0.1), (9.0, 0.2), (10.0, 0.3)])
        @test_throws ArgumentError extract(rast, [(9.0, 0.1), (0.0, missing), (9.0, 0.2), (10.0, 0.3)])
        @test_throws ArgumentError extract(rast, [(X=0.0, Y=missing), (9.0, 0.1), (9.0, 0.2), (10.0, 0.3)])
```